### ISSUE TITLE
gives an error in 8-bit mode: "ValueError: Quantization is only supported on the GPU."

### ISSUE THREAD

**Issue by nikuson** [Lightning-AI/lit-llama#123]
gives an error in 8-bit mode: "ValueError: Quantization is only supported on the GPU."  But I have a GPU connected and without quantization everything works fine

**Comment by lantiga** [comment]
Can you try specifying --accelerator "cuda" when calling generate.py?

**Comment by nikuson** [comment]
> Can you try specifying --accelerator "cuda" when calling generate.py?

`CUDAAccelerator` can not run on your system since the accelerator is not available. The following accelerator(s) is available and can be passed into `accelerator` argument of `Fabric`: ['cpu'].

**Comment by lantiga** [comment]
Can you run `nvidia-smi` on your system?

**Comment by awaelchli** [comment]
Maybe you have a torch installation without cuda. I'm pretty sure `torch.cuda.device_count()` returns 0 for you. The simplest is probably to reinstall torch. You can select the right command here: https://pytorch.org/

**Comment by nikuson** [comment]
> Maybe you have a torch installation without cuda. I'm pretty sure `torch.cuda.device_count()` returns 0 for you. The simplest is probably to reinstall torch. You can select the right command here: https://pytorch.org/

Yes, sorry, while updating pytorch accidentally installed a version without cuda
